cypress/support/hub-commands.ts

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access */
+import shell from 'shell-escape-tag';
+import './commands';
+import './rest-commands';
+import { HubItemsResponse } from '../../frontend/hub/useHubView';
+import { CollectionVersionSearch } from '../../frontend/hub/collections/CollectionVersionSearch';
+
+const apiPrefix = Cypress.env('HUB_API_PREFIX') as string;
+
+// GalaxyKit Integration: To invoke `galaxykit` commands for generating resources
+Cypress.Commands.add('galaxykit', (operation: string, ...args: string[]) => {
+  const adminUsername = Cypress.env('HUB_USERNAME') as string;
+  const adminPassword = Cypress.env('HUB_PASSWORD') as string;
+  const galaxykitCommand = Cypress.env('HUB_GALAXYKIT_COMMAND') as string;
+  const server = (Cypress.env('HUB_SERVER') as string) + apiPrefix;
+  const options = { failOnNonZeroExit: false };
+
+  cy.log(`${galaxykitCommand} ${operation} ${args.join(' ')}`);
+
+  const cmd = shell`${shell.preserve(
+    galaxykitCommand
+  )} -s ${server} -u ${adminUsername} -p ${adminPassword} ${shell.preserve(
+    operation
+  )} ${args}` as string;
+
+  cy.exec(cmd, options).then(({ code, stderr, stdout }) => {
+    cy.log(`RUN ${cmd}`, code, stderr, stdout).then(() => {
+      if (code || stderr) {
+        cy.log('galaxykit code: ' + code.toString()).then(() => {
+          cy.log('galaxykit stderr: ' + stderr).then(() => {
+            throw new Error(`Galaxykit failed: ${stderr}`);
+          });
+        });
+      }
+    });
+
+    cy.log(`stdout: ${stdout}`).then(() => {
+      return stdout.split('\n').filter((s) => !!s);
+    });
+  });
+});
+
+Cypress.Commands.add(
+  'createApprovedCollection',
+  (namespaceName: string, collectionName: string, tags?: string[]) => {
+    const waitTillPublished = (maxLoops: number) => {
```

Comment:
`galaxykit task wait all` might help .. it will start polling the api until there are no running tasks, and exit after

but this is fine too :)

(except this only works when `settings.GALAXY_REQUIRE_CONTENT_APPROVAL === false` and signing is either disabled, or GALAXY_AUTO_SIGN_COLLECTIONS or not GALAXY_REQUIRE_SIGNATURE_FOR_APPROVAL)

Reply:
I like the idea for running `galaxykit task wait all`. However, if based on the current plan the same server is utilized across several PR checks (eventually) then it sounds like that command would wait for all tasks to complete even the ones unrelated to the creation of this collection, which could add unnecessary delay. What do you think?

Should I be setting one of these flags like `GALAXY_REQUIRE_CONTENT_APPROVAL` in the settings API before we run this command?

Reply:
>  sounds like that command would wait for all tasks to complete even the ones unrelated to the creation of this collection

Oh, you're absolutely right, task wait all might take ages when multiple clients are creating new tasks. Especially if anyone decides to test a long-running sync or bulk deletions :).


> Should I be setting one of these flags like GALAXY_REQUIRE_CONTENT_APPROVAL in the settings API 

For now, we can only test the server running with default settings, the default default for PAH is that collections require approval.

Eventually we'll want to test all the modes, but ... right now there is no settings API to push to (until https://github.com/ansible/galaxy_ng/pull/1837), and such a settings change would affect all the clients anyway.. So I'm thinking we'll need multiple server instances to test all the modes.